src/internal/AudioDSPKernel.h

```diff
@@ -13,28 +13,28 @@ namespace lab {
     
 // AudioDSPKernel does the processing for one channel of an AudioDSPKernelProcessor.
 
-class AudioDSPKernel {
+class AudioDSPKernel 
+{
 public:
-    AudioDSPKernel(AudioDSPKernelProcessor* kernelProcessor);
-    AudioDSPKernel(float sampleRate);
+
+    AudioDSPKernel(AudioDSPKernelProcessor * kernelProcessor);
+    AudioDSPKernel();
+
     virtual ~AudioDSPKernel() {}
 
     // Subclasses must override process() to do the processing and reset() to reset DSP state.
     virtual void process(ContextRenderLock&, const float* source, float* destination, size_t framesToProcess) = 0;
     virtual void reset() = 0;
 
-    float sampleRate() const { return m_sampleRate; }
-    double nyquist() const { return 0.5 * sampleRate(); }
+    AudioDSPKernelProcessor * processor() { return m_kernelProcessor; }
+    const AudioDSPKernelProcessor * processor() const { return m_kernelProcessor; }
 
-    AudioDSPKernelProcessor* processor() { return m_kernelProcessor; }
-    const AudioDSPKernelProcessor* processor() const { return m_kernelProcessor; }
-
-    virtual double tailTime() const = 0;
-    virtual double latencyTime() const = 0;
+    virtual double tailTime(ContextRenderLock & r) const = 0;
+    virtual double latencyTime(ContextRenderLock & r) const = 0;
```

Comment:
it's kind of gross that only the fft has a latency time, yet the complexity of the latency time is propagated through all nodes. latencyTime is a similar gross thing, it seems like a good idea but for all intents and purposes almost nothing uses it. future @TODO...

Reply:
(I meant tailTime and latencyTime)